File: BackEnd.v2.1/demo/src/main/java/com/example/demo/models/Inverter.java
package com.example.demo.models;


import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.UUID;

@Data
@Entity
@Table(name = "inverter")
public class Inverter {
    @Id
    @GeneratedValue
    private UUID id;
    private String model;
    private Float nominal_power;
    private Float tension;
    private String frequency;
    private String type;
    private Integer phase_number;
    private Float maximum_circuit_voltage;
    private Float minimal_circuit_current;
    private Float price;
    private Integer quantity;
    @Lob
    @Column(name = "image", length = 500000000)
    private String image;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Company company;
}

File: FrontEnd.v2/src/app/models/inverter.ts
/* tslint:disable */

/* eslint-disable */
import {Company} from "./Company";

export interface Inverter {
  companyDto: Company;
  frequency: string;
  id?: any;
  image: string;
  maximum_circuit_voltage: number;
  minimal_circuit_current: number;
  model: string;
  nominal_power: number;
  phase_number: number;
  price: number;
  quantity: number;
  tension: number;
  type: string;
}
